src/setupEnzyme.test.tsx

```diff
@@ -4,6 +4,7 @@ import { setupEnzyme } from "./setupEnzyme";
 
 type MyComponentProps = {
   text: string;
+  optional?: number;
```

Comment:
curious why you made this an optional rather than a mandatory prop for your test?

Reply:
Good question! So the following is something I think I'll add to the top of the types file to ground future readers:

Users of the `setup*` methods will get a `render*` function that they'll use to test against the render/behavior of component they're testing. For these tests, there are often component props (required or optional) that are the same across tests that if repeated in each one would cause the tests to be bloated and harder to read.

In an effort to DRY them out, we use the aforementioned `setup*` helper to effectively [curry](https://en.wikipedia.org/wiki/Currying) this method to allow one of three major variants in our testing:

1) User provides all required props and wants to add nothing specific to a test:
```
const renderWrapper = setupEnzyme({ text: 'default value' });

it('does something', () => {
  const { wrapper } = renderWrapper();
  // insert test actions & assertions
});
```

Note in the above, the caller did not need to even provide an argument for the `renderWrapper` method, as all of the required props had already been provided in `setupEnzyme`.

2) User provides _some_ required props but wants to enforce that each test defines some things:
```
type MyComponentProps = {
  text: string;
  text2: string;
  optional?: number;
}
  
const renderWrapper = setupEnzyme({ text: 'default value' });

it('does something', () => {
  const { wrapper } = renderWrapper({ text2: 'specific test inclusion' });
  // insert test actions & assertions
});
```

Note that since `text2` is also required, the `renderWrapper` method will throw a compile error if it's not provided here. This is really the core magic of this PR, which properly parses what keys were passed in `setup*` and enforces that the complement of the props are provided in each test.

3) Lastly, we also want to allow callers to provide test-specific _overrides_ of their initially passed values:
```
const renderWrapper = setupEnzyme({ text: 'default value' });

it('does something', () => {
  const { wrapper } = renderWrapper({ text: 'override value' });
  // insert test actions & assertions
});
```
Just like in case (1), the caller _could_ provide nothing into the `renderWrapper` method, it's already satisfied the interface, however, we _allow_ the caller to provide the overrides if they should wish to (happens all the time). Allowing these two variants of behavior is actually the core reason for why we're using the spread and `[T?]` syntax in the types.

Anyway, this is a long answer to your question (and I'm gonna add it to the branch itself), but to answer your _specific_ question, I added optional keys to ensure that the type system wasn't incorrectly flagging optional props as required. I'll add that comment, too =)